walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,78 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+        return Arrays.asList();
+    }
+    ArrayList<TimeRange> occupiedSlotsMandatory =  new ArrayList<TimeRange>();
+    ArrayList<TimeRange> occupiedSlotsAll =  new ArrayList<TimeRange>();
+    boolean foundAnyAttendee;
+    for (Event event: events) {
```

Comment:
I'm not particularly fond of a break; if it isn't essential. Also the usage of foundAnyAttendee isn't that clear to me, please elaborate (are you trying to avoid the scenario of an attendee being both optional and mandatory?) A cleaner way to approach this IMO is to start by having your attendees separated. Before you go into the loop, I'd use Java's HashSet (https://docs.oracle.com/javase/7/docs/api/java/util/HashSet.html) to form two sets (there's a constructor that receives a collection):
1. HashSet of mandatory attendees.
2. HashSet of optional attendees.
3. Remove all occurrences of attendees in (1) from (2).

You can then quietly iterate over your events, and for all attendees do:
if (optionalAttendees.contains(attendee) || mandatoryAttendees.contains(attendee)) {
 if (mandatoryAttendees.contains(attendee)) { // Add time to occupiedSlotsMandatory}
 // Add time to occupiedSlotsAll
}

Reply:
The break is because once I found a mandatory attendee I want to stop iterating since it doesn't matter if there are more common attendees.
The use of the foundAnyAttendee is because once I found any attendee (optional or madatory) I don't want to keep checking  request.getOptionalAttendees().contains(attendee) because it takes one attendee for the slots to be blocked. So both the break and the foundAnyAttendee are for stop checking for common attendees once one is found. 
Also, I didn't want a list of just the optionals since I check for slots once for only mandatory and once for everyone (and never just for optionals).

Reply:
Is break something that is not usualy used?